src/index.ts

```diff
@@ -92,25 +110,53 @@ export class WebRTCPlayer extends EventEmitter {
 
   private onErrorHandler(error: string) {
     this.log(`onError=${error}`);
-    switch(error) {
+    switch (error) {
       case "reconnectneeded":
         this.peer && this.peer.close();
-        this.videoElement.srcObject = undefined;
+        this.videoElement.srcObject = null;
         this.setupPeer();
         this.adapter.resetPeer(this.peer);
         this.adapter.connect();
         break;
     }
   }
 
-  private async onConnectionStats() {
+  private async onConnectionStats() { 
+    
     if (this.peer && this.statsTypeFilter) {
+      let bytesReceivedBlock: number = 0;
       let stats = await this.peer.getStats(null);
+
       stats.forEach((report) => {
         if (report.type.match(this.statsTypeFilter)) {
           this.emit(`stats:${report.type}`, report);
         }
+
+        //inbound-rtp attribute bytesReceived from stats report will contain the total number of bytes received for this SSRC.
+        //In this case there are several SSRCs. They are all added together in each onConnectionStats iteration and compared to their value during the previous iteration.
+        if (report.type.match('inbound-rtp') ) { 
+          bytesReceivedBlock += report.bytesReceived;
+        }
       });
+
+      if (bytesReceivedBlock <= this.bytesReceived) {
+        this.timeoutThresholdCounter += this.msStatsInterval;
+
+        if (this.mediaTimeoutOccured == false && this.timeoutThresholdCounter >= this.mediaTimeoutThreshold) {
```

Comment:
nit-pick. It's good practice to always do `===` since `== ` doesn't check types. eg `"" == false` is true but `"" === false` is false. ( For more info, read [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) "In most cases, using loose equality is discouraged. The result of a comparison using strict equality is easier to predict, and may evaluate more quickly due to the lack of type coercion." )